src/NuGet.Core/NuGet.Protocol/Properties/AssemblyInfo.cs

```diff
@@ -10,12 +10,14 @@
 [assembly: InternalsVisibleTo("NuGet.Credentials.Test, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
 [assembly: InternalsVisibleTo("NuGet.Protocol.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
 [assembly: InternalsVisibleTo("NuGet.VisualStudio.Common.Test, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
+[assembly: InternalsVisibleTo("NuGet.CommandLine.Xplat.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```

Comment:
Assembly `X` using `InternalsVisibleTo` assembly `X.Tests` is acceptable, but giving it to another else is a code smell. I see it's being done because the json protocol model is read only, and you want to be able to set property values in a test. But customers using `NuGet.Protocol` in their own app may want to be able to test their own code, and don't have the luxury of being able to add `InternalsVisibleTo` for their own assemblies.

If it's hard to write test for in our own repo, then it's hard for customers to test, so I think the design should be improved.

Reply:
I'll look into it. Thanks.

Reply:
@zivkan It'd be good to have this conversation tracked here for future reference. I'm trying to discern the difference between what I'm doing with `NuGet.CommandLine.Xplat.Tests` and what has been done here with `NuGet.VisualStudio.Common.Test` and `NuGet.Credentials.Test`, but coming up empty. Could you fill me in, please?

Reply:
As discussed on Teams, I'm not sure why these other test projects need/use `InternalsVisibleTo`. They existed before I joined NuGet. My feedback is just that if it's useful for us to test, the customers using `NuGet.Protocol` in their own projects probably would like to be able to test as well.

I won't block the PR over this, but projects that make up the NuGet SDK are available for customers to use in their own projects, and I'd like us to be more mindful of this in the future than we have been in the past.